### PR TITLE
Adds sentiment analysis calculation to comments

### LINKED ISSUE #42: Modify your DataServlet.java to calculate the sentiment score
- [x] Modify your `DataServlet.java` to calculate the sentiment score of the comment entered by the user
- [x] Test that this works by printing it out to the `console`

### LINKED ISSUE #43: Add the sentiment score to your comment storage
- [x] Add a field in `comment.java` that stores the sentiment score

### LINKED ISSUE #44: Add the sentiment score to the JSON returned by your servlet
- [x] Add sentiment score to the `JSON`

### LINKED ISSUE #45: Add JavaScript code that displays sentiment in HTML
- [x] Add JavaScript code that reads the sentiment score from the `JSON` and displays it
- [x] Optional, translate score into emoji

### PR DESCRIPTION
**Pull request by jaysaleh**
This pull request computes the sentiment analysis score for each comment and displays it for the user to view.

Screenshots: https://screenshot.googleplex.com/50EHoUMLQ2m.png

### CODE REVIEW

File: portfolio/src/main/webapp/script/comment-script.js [hunk 1]
```diff
@@ -146,27 +147,33 @@ function createImageDiv(imageUrl) {
...
+function createCommentDiv(text, email, timeStamp, sentimentScore) {
   const commentOuterDiv = document.createElement('div');
   const commentDiv = document.createElement('div');
 
   const textElement = document.createElement('p');
   const emailElement = document.createElement('h4');
+  const scoreElement = document.createElement('h5');
   const dateElement = document.createElement('h5');
 
+  var scoreWord = sentimentScore >= 0 ? 'good' : 'bad';
```
Comment (f-camp) [inline portfolio/src/main/webapp/script/comment-script.js]: Please add a comment on what >=0 means and <0 means? What does good or bad mean, its quite vague and a bit of an explanation would help.

**Comment by f-camp** [comment]
From the screen shots, I notice that good or bad are the only two gradings of the comment's sentiment. What if we had a mixed comment like: "This site does well in representing projects but I think the commenting experience is not there yet", for example. What would be the sentiment grading for such a comment?

File: portfolio/src/main/webapp/script/comment-script.js (hunk 1 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script/comment-script.js]: fixed

**Comment by jaysaleh** [comment]
> From the screen shots, I notice that good or bad are the only two gradings of the comment's sentiment. What if we had a mixed comment like: "This site does well in representing projects but I think the commenting experience is not there yet", for example. What would be the sentiment grading for such a comment?

The sentiment score generated ranges from -1 to 1. -1 meaning the comment is most likely negative and 1 meaning the comment is most likely positive. So any comment will either be on the positive or the negative side hence my choice of using `good` and `bad`. To remedy the entirety if the comment being labeled as 'good' or 'bad' I placed a percentage before the label. Running the example you provided through the sentiment analysis I got back `30% Bad`. I think may include 2 more labels, `leans bad` and `leans good`. lmk what you think and I'll add it as a TODO.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -83,13 +87,26 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
+    commentEntity.setProperty(SENTIMENT_SCORE, getSentiment(commentText));
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(commentEntity);
 
     response.sendRedirect(REDIRECT_URL);
   }
 
+  /**
+   * Returns sentiment analysis score of {@code commentText}.
```
Comment (billie-cat) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: This method should return what the range is and means.

File: portfolio/src/main/webapp/script/comment-script.js [hunk 3]
```diff
@@ -106,7 +106,7 @@ async function populateBlobUrl() {
  * Creates and returns a <div> containing the comment
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script/comment-script.js]: should this comment be updated?

File: portfolio/src/main/webapp/script/comment-script.js [hunk 4]
```diff
@@ -146,27 +147,34 @@ function createImageDiv(imageUrl) {
...
   const commentOuterDiv = document.createElement('div');
   const commentDiv = document.createElement('div');
 
   const textElement = document.createElement('p');
   const emailElement = document.createElement('h4');
+  const scoreElement = document.createElement('h5');
   const dateElement = document.createElement('h5');
 
+  // The comment is most likely good if score is postivie. Otherwise it is most likely bad.
+  var scoreWord = sentimentScore >= 0 ? 'good' : 'bad';
```
Comment (billie-cat) [inline portfolio/src/main/webapp/script/comment-script.js]: Please make a const for the number 0. The const name can be "POSITIVE_SENTIMENT_THRESHOLD"

**Review by billie-cat — CHANGES_REQUESTED** [review]

File: portfolio/src/main/webapp/script/comment-script.js (hunk 4 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script/comment-script.js]: fixed

**Review by f-camp — APPROVED** [review]

**Comment by f-camp** [comment]
> The sentiment score generated ranges from -1 to 1. -1 meaning the comment is most likely negative and 1 meaning the comment is most likely positive. So any comment will either be on the positive or the negative side hence my choice of using good and bad. To remedy the entirety if the comment being labeled as 'good' or 'bad' I placed a percentage before the label. Running the example you provided through the sentiment analysis I got back 30% Bad. I think may include 2 more labels, leans bad and leans good. lmk what you think and I'll add it as a TODO.

I think this is fine then, the percentage added with good or bad classification gives more context.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (jaysaleh) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: fixed

File: portfolio/src/main/webapp/script/comment-script.js (hunk 3 above)
Comment (jaysaleh) [reply portfolio/src/main/webapp/script/comment-script.js]: updated to include horizontal line

File: portfolio/src/main/webapp/css/comments-style.css [hunk 5]
```diff
@@ -378,3 +386,6 @@ textarea {
   }
 }
 
+
```
Comment (billie-cat) [inline portfolio/src/main/webapp/css/comments-style.css:389]: These new lines can be removed.

**Review by billie-cat — APPROVED** [review]

**Pull request merged**